policy: Disable well-known identities for non-managed etcd

tgraf opened the pull request:
Add an option --enable-well-known-identities to allow disabling well-known
identities for all new deployments which are not using managed etcd. This
reduces the number of security identities whitelisted for each endpoint.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/9698)
<!-- Reviewable:end -->

joestringer commented on Documentation/install/upgrade.rst:
```diff
@@ -313,6 +313,26 @@ Removed options
 
+.. _1.7_upgrade_notes:
+
+1.7 Upgrade Notes
```
The section immediately above this already has the 1.7 Upgrade notes title.

tgraf replied:
Fixed

joestringer reviewed (changes requested):
Minor docs nit, otherwise LGTM.

tgraf committed:
policy: Disable well-known identities for non-managed etcd

Add an option --enable-well-known-identities to allow disabling well-known
identities for all new deployments which are not using managed etcd. This
reduces the number of security identities whitelisted for each endpoint.

Signed-off-by: Thomas Graf <thomas@cilium.io>

joestringer reviewed (approved):
Looks good. Are there any repercussions for disabling the well-known identities for kube-dns? /cc @aanm
